Integrations/integration-McAfeeNSM.yml

```diff
@@ -0,0 +1,679 @@
+    }
+
+    function get_IPS_policy_details(policy_id) {
+        var cmd_url = 'ipspolicy/{0}'.format(policy_id);
+        var res = request('GET', cmd_url, {}, '');
+
+        return IPS_policy_to_entry('IPS Policy {0}'.format(policy_id), res.PolicyDescriptor, policy_id);
+    }
+
+    function get_sensors(domain_id) {
+        var cmd_url = 'sensors';
+        if (domain_id !== undefined) {
+            cmd_url += '?domain={0}'.format(domain_id);
+        }
+        var res = request('GET', cmd_url, {}, '');
+
+        return sensors_to_entry('Sensors', res.SensorDescriptor);
+    }
+
+    function get_domains(domain_id) {
+        var cmd_url = 'domain';
+        if (domain_id) {
+            cmd_url += '/{0}'.format(domain_id);
+        }
+        // var res = request('GET', cmd_url, {}, JSON.stringify({'domain_id' : domain_id}));
+        //log(JSON.stringify(res));
+        var res = request('GET', cmd_url, {}, '');
+
+        if (domain_id) {
+            return domains_to_entry('Domain {0}:'.format(domain_id), [res]);
+        } else {
+            return domains_to_entry('{0} Domains:'.format(res.length || ''), [res]);
+        }
+    }
+
+    function all_alerts_url(state, time_period, start_time, end_time, search, filter) {
+        var cmd_url = 'alerts';
+        var cmd_args = {};
+
+        if (state) {
+            cmd_args.alertstate = state;
+        }
+
+        if (time_period) {
+            cmd_args.timeperiod = time_period;
+            if (time_period === 'CUSTOM') {
+                if ((start_time === undefined) || (end_time === undefined)) {
+                    return 'Error! In CUSTOM mode, You must specify both start time and end time';
+                }
+                cmd_args.starttime = start_time;
+                cmd_args.endtime = end_time;
+            }
+        }
+
+        if (search) {
+            cmd_args.search = search;
+        }
+
+        if (filter) {
+            cmd_args.filter = filter;
+        }
+
+        return cmd_url + encodeToURLQuery(cmd_args);
+    }
+
+    function get_alerts(state, time_period, start_time, end_time, search, filter) {
+        var cmd_url = all_alerts_url(state, time_period, start_time, end_time, search, filter);
+        var res = request('GET', cmd_url, {}, '');
+        log('Showing {0}/{1} Alerts:'.format(res.retrievedAlertsCount, res.totalAlertsCount));
+        return alerts_to_entry('Showing {0}/{1} Alerts:'.format(res.retrievedAlertsCount, res.totalAlertsCount), res.alertsList);
+    }
+
+    function update_alerts(state, time_period, start_time, end_time, search, filter, new_state, new_assignee) {
+        var cmd_url = all_alerts_url(state, time_period, start_time, end_time, search, filter);
+        if ((new_state === undefined) && (new_assignee === undefined)) {
+            return 'Error! You must specify a new alert state or a new assignee';
+        }
+
+        var query = {};
+        if (new_state !== undefined) {
```

Comment:
in JS is more common to write  `if (!new_state)`

That's ok also of course :)

Reply:
please do change